src/ocramius/util/Optional.php

```diff
@@ -44,18 +29,22 @@
  * (including reference equality ({@code ==}), identity hash code, or
  * synchronization) on instances of {@code Optional} may have unpredictable
  * results and should be avoided.
+ *
+ * @template T The type of the value residing in the Optional wrapper.
  */
 final class Optional
 {
     /**
      * Common instance for {@code empty()}.
+     *
+     * @var self
      */
     private static $EMPTY;
 
     /**
      * If non-null, the value; if null, indicates no value is present
      *
-     * @var mixed
+     * @var T
```

Comment:
How do other tools handle this parametric `T`? Is it possible to use a syntax that doesn't confuse those?

Reply:
Sorry for the late reaction. I don't think there is anything that's compatible with most other tools. Is there any tool specifically you think about? I'm afraid PHP not providing generics as a language feature makes a good solution very hard.

Reply:
The syntax `@var T` is interpreted as `@var \This\Namespace\T` by IDEs and editors with intellisense. Is there nothing similar to `@var`, but psalm-specific (to avoid the collision with the existing semantics)?

Reply:
Aye, sir. I found `@psalm-param` etc in the documentation of Psalm. Added them and reverted the other annotations back to just PHP types.

Reply:
:+1: good one!